pilot/pkg/xds/delta.go

```diff
@@ -448,11 +448,7 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, push *model.PushContext,
 		Nonce:             nonce(push.LedgerVersion),
 		Resources:         res,
 	}
-	// We take the set of watched resources and anything not in the response is sent as RemovedResources
-	// This is similar to SotW, but done on the server side instead of the client.
-	cur := sets.NewSet(w.ResourceNames...)
```

Comment:
I think you still need this code for the generators that don't have first class delta support? otherwise we never delete

Reply:
fixed